app/models/message.rb

```diff
@@ -189,11 +189,24 @@ def send_reply
     attachments.blank? ? ::SendReplyJob.perform_later(id) : ::SendReplyJob.set(wait: 2.seconds).perform_later(id)
   end
 
-  def reopen_conversation
+  def update_conversation_status
     return if conversation.muted?
     return unless incoming?
 
-    conversation.open! if conversation.resolved? || conversation.snoozed?
+    conversation.open! if conversation.snoozed?
+
+    if conversation.resolved?
```

Comment:
lets move this logic to a separate method to avoid the nested if else

Reply:
@sojan-official This is fixed.